core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -555,8 +555,12 @@ class GroupMetadataManager(brokerId: Int,
               }
               pendingOffsets.remove(batch.producerId)
             } else {
+              var batchBaseOffset: Option[Long] = None
               for (record <- batch.asScala) {
                 require(record.hasKey, "Group metadata/offset entry key should not be null")
+                if (batchBaseOffset.isEmpty) {
+                  batchBaseOffset = Some(record.offset())
+                }
```

Comment:
Nit: we don't need the block or `()` in `record.offset`.

Reply:
personally i've seen waaaay too many times how adding an innocent-looking logging statement changes control flow because a `{}` block wasnt used, but will fix :-)